datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/MutableTypeBatchResolver.java

```diff
@@ -0,0 +1,58 @@
+package com.linkedin.datahub.graphql.resolvers.mutate;
+
+import com.codahale.metrics.Timer;
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.types.MutableType;
+import com.linkedin.metadata.utils.metrics.MetricUtils;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+
+
+/**
+ * Generic GraphQL resolver responsible for performing updates against particular types.
+ *
+ * @param <I> the generated GraphQL POJO corresponding to the input type.
+ * @param <T> the generated GraphQL POJO corresponding to the return type.
+ */
+public class MutableTypeBatchResolver<I, T> implements DataFetcher<CompletableFuture<List<T>>> {
+
+  private static final Logger _logger = LoggerFactory.getLogger(MutableTypeBatchResolver.class.getName());
```

Comment:
So another option would be to just have a single <I> instead of an array.

Like BatchUpdateDatasetsInput.class as the type instead of arrayUpdateType